docs/language/contracts.rst

```diff
@@ -18,19 +18,11 @@ like so:
       }
   }
 
-A constructor does not have a name and may have any number of arguments. If a constructor has arguments,
-then when the contract is deployed then those arguments must be supplied.
+In Substrate, constructor can optionally have a name and may have any number of arguments.
+If a constructor has arguments, then when the contract is deployed then those arguments must be supplied.
```

Comment:
Your version reads like arguments are only permitted on Substrate

```suggestion
A constructor can have any number of arguments. In Substrate, it may have a name. The constructor
name is only used for the metadata, nothing further.
If a constructor has arguments, then when the contract is deployed then those arguments must be supplied.
```

Reply:
> Please can have some tests that check that constructor names cannot be the same as function names

Hmm I'm not exactly sure where to implement this check, most likely in `src/sema/contracts.rs` ?